src/gui/properties/peersadditiondlg.h

```diff
@@ -33,7 +33,7 @@
 
 #include "base/bittorrent/peerinfo.h"
 
-template <class T> class QList;
+template <typename T> class QList;
```

Comment:
Shouldn't we just include <QList> instead of forward declare it?

Reply:
Is a rule about this? Don't we do the same in other places too?

Reply:
>Is a rule about this? 

Not one I'm aware of.

>Don't we do the same in other places too?

Yeah, I guess it comes down on how much longer to compile when QList header is included, I would guess not much and also since `QList` is quite a basic type so (I think) it make sense to include it directly when it is being used.

Reply:
You're not gonna fix this here?

Reply:
It seems like it's generally a bad idea to forward declare types from 3rd party libraries (especially templates). E.g.: [Google C++ Style Guide](http://google.github.io/styleguide/cppguide#Forward_Declarations).